runtime/parachains/src/dmp.rs

```diff
@@ -223,6 +223,12 @@ impl<T: Config> Module<T> {
 	}
 }
 
+impl<T: Config> crate::ParachainCleanup for Module<T> {
```

Comment:
what's this for?

Reply:
This is to remove tight coupling between the registrar pallet and hrmp, dmp, and ump pallets.

When a parachain is deregistered, all parachain data needs to be removed at the end of the next session.

This hook calls into all the pallets which may have some data for a parachain, for example the queued messages in dmp, ump, or hrmp, and tells those pallets to clean up that data at the end of the next session.

Reply:
Ok - might be better named `OnParachainDecommissioned`